### ISSUE TITLE
Zero Knowledge Policy

### ISSUE THREAD

**Issue by Peppernrino** [moisesmcardona/DistributedDataAndMediaEncoding#2]
I think it would be attractive to implement a zero-knowledge policy or something similar. As someone who transcodes sensitive material, it would be nice to know that the people on the other end aren't snooping.

**Comment by moisespr123** [comment]
I agree. This is part of the reason my project is closed. I don't think there's an easy way to encode files without the user knowing it (Having a policy doesn't guarantee the user will not be snooping through the files, since they are easily accessible anyway). Maybe if there's a VM with the files embedded in it they can be protected. However, this would make encoding slow,l as everything would need to run inside the VM. Encoders can't access encrypted files before first uncompressing it. Currently, the way the project works is by downloading the encoding app if it isn't yet downloaded or if it's updated, and then the media file is downloaded. Afterward, the WU runs and the result file is uploaded. The validator will validate the work unit and the assimilator will make sure the resulting file is placed in the correct output folder. 

I have no idea how to make custom VM's and how to run tasks in it. 

Feel free to try making a project fork and launch your own server to do your own tests!
